XiEditor/EditViewController.swift

```diff
@@ -165,16 +167,35 @@ class EditViewController: NSViewController, EditViewDataSource, FindDelegate, Sc
         scrollView.hasHorizontalScroller = true
         scrollView.usesPredominantAxisScrolling = true
         (scrollView.contentView as? XiClipView)?.delegate = self
+
     }
 
     override func viewDidAppear() {
         super.viewDidAppear()
+        setupStatusBar()
         shadowView.setup()
         NotificationCenter.default.addObserver(self, selector: #selector(EditViewController.frameDidChangeNotification(_:)), name: NSView.frameDidChangeNotification, object: scrollView)
         // call to set initial scroll position once we know view size
         redrawEverything()
     }
 
+    func setupStatusBar() {
+        if self.unifiedTitlebar == true {
```

Comment:
we shouldn't need to do styling here. The statusbar should draw itself with the 'default style' (to be determined) by default, and if we're using the unified style we should be updating the statusbar in the same place where we update other elements (which is the `unifiedTitlebar`'s `didSet` method).

To see the advantage of this approach, notice that if you change the `unified_toolbar` setting, the statusbar isn't updated.

Similarly, if you select a dark theme the statusbar becomes unreadable.